### PR TITLE
tests: add unit tests for convert_write()

### PR DESCRIPTION
**Pull request by GregoryVds**
Coverage is not complete but still a nice improvement.

Signed-off-by: Gregory Vander Schueren <gregory.vanderschueren@tessares.net>
Change-Id: I132c41af91db660e9c23596149c291c80b1cd079

### CODE REVIEW

File: tests/check_convert_util.c [hunk 1]
```diff
@@ -260,9 +260,45 @@ START_TEST(test_convert_parse_tlvs_multiple){
 }
 END_TEST
 
-START_TEST(test_convert_write){
-	/* TODO */
-	ck_assert(1);
+START_TEST(test_convert_write_tlvs){
```
Comment (matttbe) [inline tests/check_convert_util.c:263]: (tiny detail: don't forget Uncrustify change ;-) )

File: tests/check_convert_util.c [hunk 2]
```diff
@@ -260,9 +260,45 @@ START_TEST(test_convert_parse_tlvs_multiple){
...
+		/* Cast to ssize_t is safe as tlv_orig_len < SSIZE_MAX */
+		ck_assert_msg(ret == ((ssize_t)hdr_and_tlv_len),
+		              "Should write size(Header)+size(TLV[%u]) bytes",
+		              i);
+
+		/* Verify copy TLV is a byte-per-byte copy of the original TLV */
+		for (j = 0; j < tlv_orig_len; ++j)
+			ck_assert_msg(tlv_orig[j] == tlv_copy[j],
+			              "Expected tlv_orig[%u][%u] == tlv_copy[%u][%u]",
+			              i, i, j, j);
```
Comment (matttbe) [inline tests/check_convert_util.c:297]: not `i, j, i, j`?

File: tests/check_convert_util.c (hunk 1 above)
Comment (GregoryVds) [reply tests/check_convert_util.c:263]: Sure, will do.

File: tests/check_convert_util.c (hunk 2 above)
Comment (GregoryVds) [reply tests/check_convert_util.c:297]: Good catch! Thanks

**Comment by GregoryVds** [comment]
Woops, bad manip :-) Should have opened it against Master branch, the other PR... Not used to working with github :o

**Comment by matttbe** [comment]
You should still be able to change the target in an already opened PR ;-)

**Pull request closed without merging**